interface/src/app/mlmodel/mlModelComponents.js

```diff
@@ -65,7 +65,7 @@ angular.module('adage.mlmodel.components', [
           self.isValidModel = true;
         },
         function error(err) {
-          MlModelTracker.reset();
+          MlModelTracker.init();
```

Comment:
I was wondering if it was safe to call `.init()` in an error handler, so I looked up `MLModelTracker` (should really be defined in `src/app/mlmodel`, by the way, because I had a hard time finding it in `utils.js`) and found that it is indeed safe because all it does is clear its properties. I think the name `reset` or `clear` would be more accurate, but I understand why you're using `init()`. Perhaps a simple way around this would be to actually define a `reset()` and call that from `init()`.

Anyway, it's not a huge concern since `init()` currently doesn't currently do anything unsafe, but it would become a problem down the road if we added some code there that wouldn't be good to call during an error handler. I'd recommend at minimum a comment in `MLModelTracker.init()` to guard against that.

Reply:
Good point. Part of the reason why `MlModelTracker` is located in `utils.js` is because this service is needed by almost every front-end module, and what it does is keep track of the current mlmodel, and it is not directly related to any front-end display.
 
The distinction of `init()` and `reset()` makes a lot of sense. Let me add a `reset` method which will call `init()` for now.

Reply:
Why don't we *define* the `MLModelTracker` service in `/src/app/mlmodel`, so it can be with all other `MLModel` code, and (since, as you say, it's needed everywhere) *include* it in `utils.js`?

Reply:
To be more precise: when I say "include", I mean use Angular's dependency mechanism to ensure it is available.

Reply:
Right now `mlmodel` directory has two js files: `mlModelComponents.js` and `resource.js`. Neither of them seems to be a good place for `MLModelTracker` service. `MlModelTracker` doesn't rely on any external modules, I would argue that it is more like a utility. (But I agree that its name may be misleading.) I also don't want to make this PR too complex.

Reply:
I agree that this is not an essential part of this PR... it just came up because I had trouble tracking down `MLModelTracker` during the review since it wasn't where I expected to find it. Since we have more details to discuss let's file an issue so it doesn't hold up the main point of this PR.

Reply:
Done!